python/lsst/daf/butler/queries/tree/_column_literal.py

```diff
@@ -386,9 +387,20 @@ def make_column_literal(value: LiteralValue) -> ColumnLiteral:
         case lsst.sphgeom.Region():
             return RegionColumnLiteral.from_value(value)
         case lsst.sphgeom.LonLat():
-            vec = lsst.sphgeom.UnitVector3d(value)
-            # Convert the point to a Region by representing it as a zero-radius
-            # Circle.
-            region = lsst.sphgeom.Circle(vec)
-            return RegionColumnLiteral.from_value(region)
+            return _make_region_literal_from_lonlat(value)
+        case astropy.coordinates.SkyCoord():
+            icrs = value.transform_to("icrs")
+            ra = icrs.ra.degree
+            dec = icrs.dec.degree
+            lon_lat = lsst.sphgeom.LonLat.fromDegrees(ra, dec)
```

Comment:
`astropy.coordinates.SkyCoord` can be array-valued (which I think means that `icrs.ra.degree` and `icrs.dec.degree` will come back as something numpy-based.  We should probably guard against this to make the exception raised friendlier.